lib/soapforce/client.rb

```diff
@@ -204,21 +207,21 @@ def describe_layout(sobject_type, layout_id=nil)
       response
     end
 
-    def query(soql)
-      call_soap_api(:query, {:queryString => soql})
+    def query(soql, header={})
```

Comment:
Instead of passing `header` into every method I think you can use `@headers` in `call_soap_api` and it would apply to all calls.  Do you ever find yourself needing different headers per request?